lib/url/URLParserPlugin.js

```diff
@@ -184,6 +184,86 @@ class URLParserPlugin {
 					relative
 				);
 				dep.loc = /** @type {DependencyLocation} */ (expr.loc);
+
+				// Handle prefetch/preload hints
+				if (importOptions) {
+					// Validate webpackPrefetch
+					if (
+						importOptions.webpackPrefetch !== undefined &&
+						importOptions.webpackPrefetch !== true
+					) {
+						parser.state.module.addWarning(
+							new UnsupportedFeatureWarning(
+								`\`webpackPrefetch\` expected true, but received: ${importOptions.webpackPrefetch}.`,
+								/** @type {DependencyLocation} */ (expr.loc)
+							)
+						);
+					}
+
+					// Validate webpackPreload
+					if (
+						importOptions.webpackPreload !== undefined &&
+						importOptions.webpackPreload !== true
+					) {
+						parser.state.module.addWarning(
+							new UnsupportedFeatureWarning(
+								`\`webpackPreload\` expected true, but received: ${importOptions.webpackPreload}.`,
+								/** @type {DependencyLocation} */ (expr.loc)
+							)
+						);
+					}
+
+					// Validate webpackFetchPriority
+					if (
+						importOptions.webpackFetchPriority !== undefined &&
+						(typeof importOptions.webpackFetchPriority !== "string" ||
+							!["high", "low", "auto"].includes(
+								importOptions.webpackFetchPriority
+							))
+					) {
+						parser.state.module.addWarning(
+							new UnsupportedFeatureWarning(
+								`\`webpackFetchPriority\` expected "low", "high" or "auto", but received: ${importOptions.webpackFetchPriority}.`,
+								/** @type {DependencyLocation} */ (expr.loc)
+							)
+						);
+					}
+
+					// Validate webpackPreloadAs
+					if (
+						importOptions.webpackPreloadAs !== undefined &&
+						typeof importOptions.webpackPreloadAs !== "string"
+					) {
+						parser.state.module.addWarning(
+							new UnsupportedFeatureWarning(
+								`\`webpackPreloadAs\` expected a string, but received: ${importOptions.webpackPreloadAs}.`,
+								/** @type {DependencyLocation} */ (expr.loc)
+							)
+						);
+					}
+
+					// Warn if both prefetch and preload are specified
+					if (
+						importOptions.webpackPrefetch !== undefined &&
+						importOptions.webpackPreload !== undefined
+					) {
+						parser.state.module.addWarning(
+							new CommentCompilationWarning(
+								"Both webpackPrefetch and webpackPreload are specified. " +
+									"webpackPreload will take precedence for immediate loading.",
+								/** @type {DependencyLocation} */ (expr.loc)
+							)
+						);
+					}
```

Comment:
Avoid it, we don't use this in other places, we need to discuss about it and if we want to validate this we should add this to all places (i.e. `import()`, `Worker`, `require()`, etc), because there are other comments for other comments that override each other

Reply:
> because there are other comments for other comments that override each other

I don't understand what this means.

When both `prefetch` and `preload` are specified like in this case, what kind of link do you think would be ideal to generate?